pkg/karmadactl/util/bootstraptoken/bootstraptoken.go

```diff
@@ -320,6 +321,8 @@ func UpdateOrCreateToken(client kubeclient.Interface, failIfExists bool, token *
 	secret, err := client.CoreV1().Secrets(metav1.NamespaceSystem).Get(context.TODO(), secretName, metav1.GetOptions{})
 	if secret != nil && err == nil && failIfExists {
 		return fmt.Errorf("a token with id %q already exists", token.Token.ID)
+	} else if err != nil && !apierrors.IsNotFound(err) {
```

Comment:
The other errors will be checked in `cmdutil.CreateOrUpdateSecret()`, it seems only preflight for `karmadactl token create` command and not to influence the CreateOrUpdateSecret phase.

And it refers to https://github.com/kubernetes/kubernetes/blob/cde45fb161c5a4bfa7cfe45dfd814f6cc95433f7/cmd/kubeadm/app/phases/bootstraptoken/node/token.go#L38

Reply:
Yes, so I think this is an optimization feature rather than a bug.

Reply:
@lonelyCZ https://github.com/kubernetes/kubernetes/pull/115000 kubeadm someone approved of my idea

Reply:
Right, if the err is not nil, the `already exist` error will be skipped that could cause the unexpected result. This patch make it safer.